container_ci_suite/engines/s2i_container.py

```diff
@@ -464,49 +465,62 @@ def doc_content_old(self, strings: List) -> bool:
     # }
 
     def test_response(
-            self, url: str = "",
-            expected_code: int = 200, port: int = 8080,
-            expected_output: str = "", max_tests: int = 20
+            self, url: str = "", port: int = 8080,
+            expected_code: int = 200, expected_output: str = "",
+            max_tests: int = 20
     ) -> bool:
-        url = f"{url}:{port}"
-        print(f"URL address to get response from container: {url}")
-        cmd_to_run = "curl --connect-timeout 10 -k -s -w '%{http_code}' " + f"{url}"
-        # Check if application returns proper HTTP_CODE
-        print("Check if HTTP_CODE is valid.")
-        for count in range(max_tests):
+        """
+        Test HTTP response of specified url
+
+        If expected output is empty, check will be skipped
```

Comment:
It does not make sense to call a response in case we do not want to check.
We always want to check an output to know the connection works properly.

Reply:
Imagine that we use some external library (which we do in case of ruby, for example) as a test sample app. We might want to check that the app works - code 200 is returned, and content might be irrelevant for us as it can change without our knowledge.

I feel like having an option to pass the test only based on correct HTTP response code ignoring content is worth having (and even preferred for test stability).

> to know the connection works properly

If it is not, you won't get 200 HTTP code.

This feature of API is optional, so it is up to test developer to correctly assess whether checking response body is needed.